### ISSUE TITLE
Low int8 accuracy after fine tuning BERT-XNLI 

### ISSUE THREAD

**Issue by masahi** [openvinotoolkit/nncf#968]
Hi, I followed the instruction in https://github.com/openvinotoolkit/nncf/tree/develop/third_party_integration/huggingface_transformers#bert-xnli to QAT and evaluate int8 BERT-XNLI. The README says the accuracy should be 77.22%, but my result is far worse than that: it is only 33% as shown in the output below.

```
***** eval metrics *****
  epoch                   =        4.0
  eval_accuracy           =     0.3333
  eval_loss               =     1.0988
  eval_runtime            = 0:06:18.44
  eval_samples            =       2490
  eval_samples_per_second =       6.58
  eval_steps_per_second   =       6.58
```

In contrast, the float model, trained with fp16 precision, gives good accuracy:
```
***** eval metrics *****
  epoch                   =        4.0
  eval_accuracy           =     0.7614
  eval_loss               =     0.7525
  eval_runtime            = 0:00:15.91
  eval_samples            =       2490
  eval_samples_per_second =    156.443
  eval_steps_per_second   =    156.443
```

I'm wondering what is wrong in my attempt. The only change I made is to reduce the batch size from 48 to 24. My PyTorch version is 1.9.1, my local nncf install is at commit https://github.com/openvinotoolkit/nncf/commit/1bd52822dec121a7c7ffaab7364e21232dd07ef4, and `transformers` is at commit `bff1c71e84e392af9625c345f9ea71f7b6d75fb3` as specified in the README.

**Comment by masahi** [comment]
> I used PyTorch 1.8.1 and NNCF from latest develop branch tip ([dc975f6](https://github.com/openvinotoolkit/nncf/commit/dc975f6207fe698b84f69043f73758c3f1f88cfe)). Which versions are you using?

My PyTorch version is 1.9.1, my local nncf install is at commit 1bd5282. I'll try PT 1.8.1, thanks.

**Comment by masahi** [comment]
@vshampor Interestingly, with PT 1.8 I got the expected int8 accuracy:
```
***** eval metrics *****
  eval_accuracy           =     0.7518
  eval_loss               =     0.6544
  eval_runtime            = 0:06:22.93
  eval_samples            =       2490
  eval_samples_per_second =      6.502
  eval_steps_per_second   =      6.502
```

So this seems like a PT 1.9 specific issue. Should I leave it open?
